src/k8s_charm.py

```diff
@@ -68,6 +81,18 @@ def on_pebble_ready(self, _) -> None:
             )
         self._update_status()
 
+    def metrics_rules(self):
```

Comment:
why not make this an instance attribute set in __init__?
```
        self._scrape = MetricsEndpointConsumer(self)
        self.metrics_rules=self._scrape.alerts
```

Reply:
feels like a lot of code to proxy a few methods.

Reply:
I think it is a bit more clear what is going on when you have virtual methods in the base class. I'm not completely apposed to this approach though

Reply:
to be discussed as a team and eventually addressed in a separate PR, don't consider this a blocker as it's purely aesthetic